lib/blackout_updater.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+class BlackoutUpdater
+  # Service Object to update blackouts in the blackouts controller
+  def initialize(blackout:, params:)
+    @blackout = blackout
+    @params = params
+  end
+
+  def update!
+    return { result: nil, error: error } if error
+    update_handler
+  end
+
+  private
+
+  attr_reader :blackout, :params
+
+  def error
+    return 'Invalid blackout parameters. Please try again.' unless
+      @blackout.validate
```

Comment:
also, this wasn't caught by any tests -- there should be a controller-level test that checks the response to attempting to update a blackout with invalid attributes